tcl/pdtk_canvas.tcl

```diff
@@ -29,31 +29,54 @@ namespace eval ::pdtk_canvas:: {
 #winfo rooty . returns contentsTop
 #winfo rootx . returns contentsLeftEdge
 
+if {$::tcl_version < 8.5 || \
+        ($::tcl_version == 8.5 && \
+             [tk windowingsystem] eq "aqua" && \
+             [lindex [split [info patchlevel] "."] 2] < 13) } {
```

Comment:
One issue is that the patch level string can contain alpha numerical values ala "8.7.a1". scan "%d.%d.%d" catches this and skips setting the value, so setting the value to 0 ahead of time yields a default of 0 even when number can't be parsed.

Reply:
I guess what I mean to say is I'm not sure how a resulting "string: < 13 comparison will play out, possibly yielding a bug where test versions of Tk will suddenly wrap the screen coords while release versions will not. It's an edge case, sure, but *could* happen ... ?

Reply:
Doh. Actually this is fine. The version of Tk I am testing with yields "8.7.a0" and the screen wrapping is not being called.

Reply:
i think, tcl is not really performing a numeric comparision with the `<` operator, but rather a string comparision.
e.g. `"abc" < "xyz"` is valid (and returns `true`). so we are safe on the syntactic side.

whether `a19 < 19` should be true or not (on the semantic side), i cannot say.